C-Plus-Plus/other/divisors_of_natural_number.cpp

```diff
@@ -0,0 +1,93 @@
+/* You are given a natural number, for which you have to find out all the possible distinct divisors of the given natural number.
+   The output is always sorted in ascending order.
+   For this we can have various approaches but the most optimized solution has a time complexity of O(sqrt(N)), and a space complexity of O(1).
+*/
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+//function to print the possible divisors of the given number.
+void divisorsOfNumber(int num)
+{
+    //iterating starting from 1 to the number itself searching for the numbers that exactly divide the given natural number, dividing it into two parts.
+    //this is done to minimize the complexity
+
+    for(int pass = 1; pass*pass < num; pass++)
+    {
+        //if the pass value exactly divides the given natural number, then it is printed
+        if(num % pass == 0)
+        {
+            cout << pass << " ";
+        }
+    }
+
+    for(int pass = sqrt(num); pass >= 1; pass--)
+    {
+        //if the pass value exactly divides the given natural number, then it is printed
+        if(num % pass == 0)
+        {
+            cout << (num/pass) << " ";
+        }
+    }
+
+    //providing a line break
+    cout << endl;
+}
+
+//main function
+int main(){
+
+    ios_base::sync_with_stdio(false);
+	cin.tie(NULL);
+#ifndef ONLINE_JUDGE
+    freopen("in.txt", "r", stdin);
+    freopen("output.txt", "w", stdout);
+#endif
```

Comment:
Improper indentation. And why are we using `txt` files for our test-cases?

Reply:
Sir most of the time I first write the code on IDE practice C++ platform where I have been learning CP, so sir when I got the question I first i wrote the program over there and then i implemented that same code which i had written there on vscode. Sir there in that site I always use that part of code because it always takes less time complexity for using cout or cin with that part of code. 
Sorry sir if I have commited some mistake but sir I have not at all copied the code from anywhere. I have implemented it on myself sir.
Sir I have made the required changes as said by you so plz sir review that and let me know.
hope my code stand good..